host_core/lib/host_core/control_interface/server.ex

```diff
@@ -586,4 +569,9 @@ defmodule HostCore.ControlInterface.Server do
       error: error
     })
   end
+
+  # returns true if all keys have non-nil values in the map
+  defp has_values(m, keys) when is_map(m) and is_list(keys) do
+    Enum.all?(keys, &Map.get(m, &1))
+  end
```

Comment:
It might be worth it to have a fallback case that just returns false, just in case we screw up the typing.

Either that, or, a dialyzer `@spec` so that we get warnings when we violate the typing, since this function is called from contexts where we are pretty sure of the data shape